tests/all/js_globals/JsString.rs

```diff
@@ -30,3 +30,39 @@ fn slice() {
         "#)
         .test()
 }
+
+#[test]
+fn substr() {
+    project()
+        .file("src/lib.rs", r#"
+            #![feature(proc_macro, wasm_custom_section)]
+
+            extern crate wasm_bindgen;
+            use wasm_bindgen::prelude::*;
+            use wasm_bindgen::js;
+
+            #[wasm_bindgen]
+            pub fn create_substr(this: &js::JsString, start: i32, length: i32) -> js::JsString {
+                this.substr(start, length)
+            }
+        "#)
+        .file("test.ts", r#"
+            import * as assert from "assert";
+            import * as wasm from "./out";
+
+            export function test() {
+                let aString = "Mozilla";
+
+                assert.equal(wasm.create_substr(aString, 0, 1), "M");
+                assert.equal(wasm.create_substr(aString, 1, 0), "");
+                assert.equal(wasm.create_substr(aString, -1, 1), "a");
+                assert.equal(wasm.create_substr(aString, 1, -1), "");
+                // TODO: Uncomment and test these assertions, once we have support for optional parameters
+                // assert.equal(wasm.create_substr(aString, -3), "lla");
```

Comment:
I did add those tests though(for the optional parameter) and do want to know what should be added as a todo if needed at all.